logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterClientUseCase.kt

```diff
@@ -100,6 +102,8 @@ class RegisterClientUseCaseImpl(
                         when {
                             failure.kaliumException.isTooManyClients() -> RegisterClientResult.Failure.TooManyClients
                             failure.kaliumException.isMissingAuth() -> RegisterClientResult.Failure.InvalidCredentials
+                            //TODO: Handle in proper way when the backend add related error for registering push token
+                            failure.kaliumException.isNotFound() -> RegisterClientResult.Failure.PushTokenRegister
```

Comment:
this solution will fail in case any of the steps before registerToken token resulted to 404 E.g. createMLSClient -> 404 reponse -> PushTokenRegister

there are 2 solutions that come to mind:

1. move registerToken to it's own usecase and clients need to call it after registering a new client.
2. handling errors for each network call in this usecase separately.

Reply:
E.G. 
```kotlin
    private suspend fun registerToken(client: Client, senderId: String): Either<RegisterClientResult.Failure, Client> = // do stuff and map to the correct error 


invoke(): RegisterClientResult {
val client = registerToken.onFailure { return it} // in case of error it will return the error and in case of success it will continue
}
```